flocker/node/agents/shadow_bind_mounter.py

```diff
@@ -0,0 +1,31 @@
+# -*- test-case-name: flocker.node.agents.test.test_shadow_bind_mounter -*-
+# Copyright ClusterHQ Inc.  See LICENSE file for details.
+
+from .blockdevice_manager import Permissions
+
+
+def create_tmpfs_shadow_mount(backing_directory, read_only_directory,
+                              blockdevice_manager):
+    """
+    Creates a "shadow mount directory". This is a tmpfs mount that is bind
+    mounted read-only to another location. The intent is to have the following
+    characteristics:
+
+        - The read_only_directory is cleared on reboots.
+        - The read_only_directory can be changed only by agents that know the
+            path of the backing directory.
+
+    :param FilePath backing_directory: The directory that will be created if
+        need be to back the shadow mount. This will be read write.
```

Comment:
Nit: I think split the first sentence in two: "The directory that backs the shadow mount. Will be created if it doesn't exist. This will be read write."